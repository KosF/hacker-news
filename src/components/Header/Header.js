import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { getStoriesList } from "Src/store/StoriesList/storiesListActions";

export const Header = ({ refreshStoriesList }) => {
  const onRefresh = () => {
    refreshStoriesList();
  };

  return (
    <header
      id="header"
      className="border-bottom mb-4 pt-2 pb-2 bg-white shadow-sm"
    >
      <div className="container d-flex justify-content-between align-items-center">
        <strong>
          <img
            src="https://i.ya-webdesign.com/images/hacker-news-logo-png-15.png"
            width="120px"
            height="51px"
            alt="Hacker News"
          />
        </strong>

        <button type="button" className="btn btn-info" onClick={onRefresh}>
          Refresh
        </button>
      </div>
    </header>
  );
};

const mapDispatchToProps = { refreshStoriesList: getStoriesList };

Header.propTypes = {
  refreshStoriesList: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Header);
