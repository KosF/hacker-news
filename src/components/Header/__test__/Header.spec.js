import React from "react";

import { Header } from "../Header";

describe("<Header />", () => {
  const props = {
    refreshStoriesList: jest.fn(),
  };

  const wrapper = global.shallow(<Header {...props} />);

  it("should render correct header", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call mock function when the 'Refresh' button is clicked", () => {
    wrapper.find("button.btn-info").simulate("click");

    expect(props.refreshStoriesList).toHaveBeenCalled();
  });
});
