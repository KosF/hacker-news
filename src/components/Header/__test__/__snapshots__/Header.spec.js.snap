// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correct header 1`] = `
<header
  className="border-bottom mb-4 pt-2 pb-2 bg-white shadow-sm"
  id="header"
>
  <div
    className="container d-flex justify-content-between align-items-center"
  >
    <strong>
      <img
        alt="Hacker News"
        height="51px"
        src="https://i.ya-webdesign.com/images/hacker-news-logo-png-15.png"
        width="120px"
      />
    </strong>
    <button
      className="btn btn-info"
      onClick={[Function]}
      type="button"
    >
      Refresh
    </button>
  </div>
</header>
`;
